src/Microphysics1M.jl

```diff
@@ -46,6 +46,35 @@ export accretion_snow_rain
 export evaporation_sublimation
 export snow_melt
 
+"""
+    logistic_function_integral(x, x_0, k)
+ 
+ - `x` - independent variable
+ - `x_0` - threshold value for x
+ - `k` - growth rate of the logistic function, characterizing steepness of the transition
+
+Returns the value of the indefinite integral of the logistic function, for smooth transitioning
+of piecewise linear profiles at thresholds. This curve smoothly transition from y = 0 
+for 0 < x < x_0 to y = x - x_0 for x_0 < x.
+"""
+function logistic_function_integral(x::FT, x_0::FT, k::FT) where {FT <: Real}
+
+    @assert k > 0
+    @assert x_0 >= 0
+    x = max(0, x)
+
+    # translation of the curve in x and y to enforce zero at x = 0
+    _trnslt = -log(1 - exp(-k)) / k
+
+    # When both x and x_0 are zero t is nan, and need to be replaced by Inf so that 
+    # the result for x = 0 is zero.
+    _t = isnan(x / x_0) ? Inf : x / x_0 - 1 + _trnslt
```

Comment:
```suggestion
    _t = isnan(x / x_0) ? FT(Inf) : x / x_0 - 1 + _trnslt
```

`Inf` is a `Float64` by default, so we'll need to wrap it in `FT` make it type-stable (in the case that we use it for calculations).

Won't `_t = Inf` result in `_result = Inf`? (is that okay?) If so, we could short-circuit to return `Inf` to avoid calling `log` and `exp` on `Inf` (`exp` is *exp*ensive).

Reply:
_t = Inf results in _kt = Inf so _kt > 50 and x-x0 is returned. For large values of _kt instead of computing log and exp I directly return x-x0 to avoid extra computations and also to avoid overflow.